client/post-editor/editor-ground-control/index.jsx

```diff
@@ -167,10 +168,13 @@ export class EditorGroundControl extends PureComponent {
 	onPreviewButtonClick = event => {
 		if ( this.isPreviewEnabled() ) {
 			this.props.onPreview( event );
-			const eventLabel = postUtils.isPage( this.props.page )
+			const googleLabel = postUtils.isPage( this.props.page )
 				? 'Clicked Preview Page Button'
 				: 'Clicked Preview Post Button';
-			recordEvent( eventLabel );
+			const tracksLabel = postUtils.isPage( this.props.page )
+				? 'calypso_editor_page_preview_button_click'
+				: 'calypso_editor_post_preview_button_click';
```

Comment:
so actually since `ownProps` is available to us from inside of `mapDispatchToProps` we don't actually have to expose these intricacies of the analytics inside the components…

```js
const mapDispatchToProps = ( dispatch, { page } ) => ( {
	recordPreviewButtonClick: () => dispatch( composeAnalytics(
		recordTracksEvent( `calypso_editor_${ postUtils.isPage( page ) ? 'page' : 'post' }_preview_button_click` ),
		recordGoogleEvent( 'Editor', `Clicked Preview ${ postUtils.isPage( page ) ? 'Page' : 'Post' } Button` )
	) )
} )
```

Reply:
Oooh, like it.

Reply:
of course, it can be valuable to have those full strings as literals and not as generated strings (for searching purposes).

```js
const mapDispatchToProps = ( dispatch, { page } )  => {
	const isPage = postUtils.isPage( page );
	const tracksLabel = …
	const googleLabel = …

	return {
		recordPreviewButtonClick: () => dispatch( composeAnalytics(
			recordTracksEvent( tracksLabel ),
			recordGoogleEvent( 'Editor', googleLabel )
		) )
	}
}
```

Reply:
good call! that's quite elegant. changing that up!